tests/Microsoft.Identity.Test.Core.UIAutomation/IOSXamarinUITestController.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xamarin.UITest.Queries;
+
+namespace Microsoft.Identity.Test.UIAutomation.Infrastructure
+{
+    public class IOSXamarinUITestController : XamarinUITestController
+    {
+        protected override void Tap(string elementID, XamarinSelector xamarinSelector, TimeSpan timeout)
+        {
+            WaitForElement(elementID, xamarinSelector, timeout);
+
+            switch (xamarinSelector)
+            {
+                case XamarinSelector.ByAutomationId:
+                    Application.Tap(x => x.Marked(elementID));
+                    break;
+                case XamarinSelector.ByHtmlIdAttribute:
+                        Application.Query(InvokeTapByCssId(elementID));
+                    break;
+                case XamarinSelector.ByHtmlValue:
+                    Application.Query(InvokeTapByHtmlElementValue(elementID));
+                    break;
+                default:
+                    throw new NotImplementedException("Invalid enum value " + xamarinSelector);
+            }
+        }
+
+        protected override void EnterText(string elementID, string text, XamarinSelector xamarinSelector, TimeSpan timeout)
+        {
+            WaitForElement(elementID, xamarinSelector, timeout);
+
+            switch (xamarinSelector)
+            {
+                case XamarinSelector.ByAutomationId:
+                    Application.Tap(x => x.Marked(elementID));
+                    Application.ClearText();
+                    Application.EnterText(x => x.Marked(elementID), text);
```

Comment:
FYI...i heard from my new app center friend that iOS has a race condition if using EnterText while already selected on a text box...not sure if this is the case here, but putting it on your radar in case we run into this at some point. :)